### ISSUE TITLE
Viz & Discovery Page Autocomplete

### ISSUE THREAD

**Issue by Hagaygur** [opensearch-project/OpenSearch-Dashboards#1176]
I am copying the [original issue](https://github.com/opendistro-for-elasticsearch/sample-code/issues/131)  , as it has since gotten no traction or issues in the OpenSearch GitHub, but seems to make life really painful for quite a whole lot of users (sorry for the formatting,trying to preserve the original issue's format while working):

Kibana OSS 6.0 - 6.3 included autocomplete, but it was removed from Kibana OSS as of 6.4.

We should consider our own implementation (presumably as a new plugin) to improve the day-to-day user experience:
[Autocomplete on discovery tab not working](https://github.com/opendistro-for-elasticsearch/sample-code/issues/86) 
[https://discuss.opendistrocommunity.dev/t/kibana-autocomplete/1779](https://discuss.opendistrocommunity.dev/t/kibana-autocomplete/1779) 
End of copy.

The issue is CRIPPLING in comparison to Kibana, especially with long and nested field names.

It really deters users, especially with the current filter tab not properly showing field names.

As a stoogap, making the filter show long field names would be nice, but is only a stopgap.

**Comment by jkowall** [comment]
Are you going to work on this @Hagaygur or are you just making a suggestion (which you have made before based on the other threads). I believe it's pretty obvious that this is a good feature to add, but it also introduces performance issues. Someone needs to sign up for doing the work.

**Comment by Hagaygur** [comment]
We've unfortunately hit a legal wall as far as contributions ATM - we have an interior patch for the filter window in our deployment, but it appears we're prevented from actually adding any code for now.

**Comment by Hagaygur** [comment]
Hey, the change is also relevant to Autocomplete on the visualizations tab

**Comment by seanneumann** [comment]
This feels like an important but not urgent capability. We definitely need it, but let's keep it on the backlog for now as we tackle some of our higher prioritized work (e.g. security, tech debt, revamping ux).

**Comment by AMoo-Miki** [comment]
@Hagaygur to help the contributors understand the scope of the feature request and the depth of user pain, may i suggest a screenshot or video?

**Comment by GN93** [comment]
Are there any updates regarding this feature or is there any workaround?
Our company is migrating from Kibana to OpenSearch and I suppose that lack of autocomplete feature might be painful for the users.
Thank you in advance for reply!

**Comment by tr0k** [comment]
We would be interested in having this feature as for some of our users it seems to be very critical. Thank you for any updates on this.

**Comment by shanilpa** [comment]
@ashwin-pc @dagneyb can we prioritize this for the next release or the one following - looks like it has been a requested enhancement for a while.

**Comment by shanilpa** [comment]
@Hagaygur, @GN93, @tr0k are there any specific use cases or workflows you can share with us so we can implement a solution that meets your needs?

**Comment by tr0k** [comment]
Hi @shanilpa, we have in mind the same as in Kibana. Please check attached screenshot.
![Uploading image (1).png…]()

**Comment by Hagaygur** [comment]
> @Hagaygur, @GN93, @tr0k are there any specific use cases or workflows you can share with us so we can implement a solution that meets your needs?

Hi, there are two major use cases:
1. Many of the users don't know the exact field names by heart, or the fields can be quite long to type out. The ability to have them suggested really helps to type them correctly and avoid boilerplate.
2. KQL (ODQL?) is hard for beginners (and even for me sometimes!). The autocomplete suggestions make it much easier to understand.

**Comment by Tabis1983** [comment]
Please add this feature. It is too difficult for the users to always guess the fieldnames and therefore really destroys the user expierience in Discover / Visualization / Dashboard.

**Comment by himanshuramchandanisonos** [comment]
This feature is very important esp for the cases where we have 100s of indices with each index having more than 50 fields. Any update on the timeline for this feature ?

**Comment by himanshur04** [comment]
This feature is very important esp for the cases where we have 100s of indices with each index having more than 50 fields. Any update on the timeline for this feature ?

**Comment by evilpanda009** [comment]
This feature would be a huge enhancement to the end user experience. Is this under progress or stuck?

**Comment by zxbuzzy** [comment]
this feature will improve the query writing experience in a very serious way
looking forward to it!)

**Comment by farukomerballi** [comment]
Yes, such a feature would really make our work much easier. We have indexes with 1200-1300 fields, which are very difficult to keep in mind.
